src/analysis/normed_space/inner_product.lean

```diff
@@ -1523,115 +1523,95 @@ end
     have ho :
-      ∀ w ∈ K, ⟪x + y - (orthogonal_projection_fn h x + orthogonal_projection_fn h y), w⟫ = 0,
+      ∀ w ∈ K, ⟪x + y - (orthogonal_projection_fn K x + orthogonal_projection_fn K y), w⟫ = 0,
     { intros w hw,
-      rw [add_sub_comm, inner_add_left, orthogonal_projection_fn_inner_eq_zero h _ w hw,
-          orthogonal_projection_fn_inner_eq_zero h _ w hw, add_zero] },
-    rw eq_orthogonal_projection_fn_of_mem_of_inner_eq_zero h hm ho
+      rw [add_sub_comm, inner_add_left, orthogonal_projection_fn_inner_eq_zero _ w hw,
+          orthogonal_projection_fn_inner_eq_zero _ w hw, add_zero] },
+    ext,
+    simp [eq_orthogonal_projection_fn_of_mem_of_inner_eq_zero hm ho]
   end,
   map_smul' := λ c x, begin
-    have hm : c • orthogonal_projection_fn h x ∈ K :=
-      submodule.smul_mem K _ (orthogonal_projection_fn_mem h x),
-    have ho : ∀ w ∈ K, ⟪c • x - c • orthogonal_projection_fn h x, w⟫ = 0,
+    have hm : c • orthogonal_projection_fn K x ∈ K :=
+      submodule.smul_mem K _ (orthogonal_projection_fn_mem x),
+    have ho : ∀ w ∈ K, ⟪c • x - c • orthogonal_projection_fn K x, w⟫ = 0,
     { intros w hw,
-      rw [←smul_sub, inner_smul_left, orthogonal_projection_fn_inner_eq_zero h _ w hw, mul_zero] },
-    rw eq_orthogonal_projection_fn_of_mem_of_inner_eq_zero h hm ho
+      rw [←smul_sub, inner_smul_left, orthogonal_projection_fn_inner_eq_zero _ w hw, mul_zero] },
+    ext,
+    simp [eq_orthogonal_projection_fn_of_mem_of_inner_eq_zero hm ho]
   end }
 
-/-- The orthogonal projection onto a subspace, which is expected to be
-complete.  If the subspace is not complete, this uses the identity map
-instead. -/
-def orthogonal_projection (K : submodule 𝕜 E) : linear_map 𝕜 E E :=
-if h : is_complete (K : set E) then orthogonal_projection_of_complete h else linear_map.id
-
-/-- The definition of `orthogonal_projection` using `if`. -/
-lemma orthogonal_projection_def (K : submodule 𝕜 E) :
-  orthogonal_projection K =
-    if h : is_complete (K : set E) then orthogonal_projection_of_complete h else linear_map.id :=
-rfl
-
 @[simp]
-lemma orthogonal_projection_fn_eq {K : submodule 𝕜 E} (h : is_complete (K : set E)) (v : E) :
-  orthogonal_projection_fn h v = orthogonal_projection K v :=
-by { rw [orthogonal_projection_def, dif_pos h], refl }
-
-/-- The orthogonal projection is in the given subspace. -/
-lemma orthogonal_projection_mem {K : submodule 𝕜 E} (h : is_complete (K : set E)) (v : E) :
-  orthogonal_projection K v ∈ K :=
-begin
-  rw ←orthogonal_projection_fn_eq h,
-  exact orthogonal_projection_fn_mem h v
-end
+lemma orthogonal_projection_fn_eq {K : submodule 𝕜 E} [complete_space K] (v : E) :
+  orthogonal_projection_fn K v = orthogonal_projection K v :=
+rfl
 
 /-- The characterization of the orthogonal projection.  -/
 @[simp]
-lemma orthogonal_projection_inner_eq_zero (K : submodule 𝕜 E) (v : E) :
+lemma orthogonal_projection_inner_eq_zero {K : submodule 𝕜 E}  [complete_space K] (v : E) :
```

Comment:
Maybe the two spaces before `[complete_space K]` here are intended to be one space?